src/testresult.cpp

```diff
@@ -0,0 +1,91 @@
+    , m_result(Result::NoResult)
+{
+    if (this->possiblyAvailable()) {
+        SFSecretHelper::Response res = SFSecretHelper::getInstance().get(TESTRESULT_KEY);
+        if (res.returncode == 0) {
+            m_result = static_cast<Result>(res.data.toInt());
+        } else {
+            qWarning() << "Couldn't retrieve testresult token from Sailfish Secrets Storage!";
+            //Change it to changed so that the initial value NoResult gets stored on destruction
+            m_result_changed = true;
+        }
+    }
+}
+
+TestResult::~TestResult()
+{
+    if (m_result_changed) {
+        if (SFSecretHelper::getInstance().remove(TESTRESULT_KEY).returncode != 0) {
+            qWarning() << "Couldn't delete old Testresult out of Sailfish Secrets Storage!";
+        }
+        if (SFSecretHelper::getInstance().store(TESTRESULT_KEY, QByteArray::number(static_cast<int>(m_result))).returncode != 0) {
+                qWarning() << "Couldn't store current Testresult in Sailfish Secrets Storage!";
+        }
+    }
+}
+
+void TestResult::startCheck()
+{
+    if (TestResult::possiblyAvailable()) {
+        if (AppSettings::getInstance().regTokenreceived().daysTo(QDate::currentDate()) > 21) {
+            m_result = Result::NoResult;
+            emit testResultChanged();
+            int returncode;
+            returncode = SFSecretHelper::getInstance().remove(REGTOKEN_KEY).returncode;
+            returncode += SFSecretHelper::getInstance().remove(TESTRESULT_KEY).returncode;
+            if (returncode == 0) {
+                AppSettings::getInstance().setRegTokenreceived(QDate());
+                if (SFSecretHelper::getInstance().store(TESTRESULT_KEY, QByteArray::number(static_cast<int>(m_result))).returncode != 0) {
+                    qWarning() << "Couldn't store current Testresult in Sailfish Secrets Storage!";
+                    m_result_changed = true;
+                }
+            } else {
+                qDebug() << "Couldn't delete expired registry token and testresult from Sailfish Secrets Storage!";
+                m_result_changed = true;
+            }
+            return;
+        }
+        if (m_result == Result::NoResult || m_result == Result::Pending) {
+            SFSecretHelper::Response res = SFSecretHelper::getInstance().get(REGTOKEN_KEY);
+            if (res.returncode == 0) {
+                emit testResultRequested(QString::fromLatin1(res.data));
+            } else {
+                qDebug() << "Couldn't retrieve registry token from Sailfish Secrets Storage!";
+            }
+        }
+    }
+}
+
+QString TestResult::testClassLabel() const
+{
+    return m_resultLabels.at(static_cast<unsigned int>(m_result));
+}
+
+bool TestResult::possiblyAvailable() const
+{
+    return AppSettings::getInstance().regTokenreceived().year() != -1;
+}
+
+TestResult::Result TestResult::currentResult() const
+{
+    return m_result;
+}
+
+void TestResult::onTestResultDownloaded(const QString &result)
```

Comment:
This currently isn't used anywhere, is that correct? It looks like maybe `Upload::onCheckForTestResultFinished` should be hooked up to this?

Reply:
I accidentally deleted the connection. I've fixe it now locally.